.github/workflows/run-dev-tests.yml

```diff
@@ -102,6 +102,7 @@ jobs:
         with:
           name: coverage-data-${{ runner.os }}-py${{ matrix.py }}-${{ matrix.version }}
           path: .coverage*
+          include-hidden-files: true
```

Comment:
> Also, there seems to be a problem in CI where the coverage data (`.coverage*` files) isn't generated into repo root. This doesn't happen locally, so I'll have to investigate more in depth.

This was a problem with `actions/upload-artifact`'s new input `include-hidden-files` which defaults to `false`. Setting this to `true` fixed the uploading of coverage data as artifacts.